impeller/entity/entity.cc

```diff
@@ -99,4 +102,64 @@ bool Entity::Render(const ContentContext& renderer,
   return contents_->Render(renderer, *this, parent_pass);
 }
 
+Vector4 min(Vector4 value, float threshold) {
+  return Vector4(std::min(value.x, threshold), std::min(value.y, threshold),
+                 std::min(value.z, threshold), std::min(value.w, threshold));
+}
+
+Color blendColor(const Color& src,
```

Comment:
Not an expert about rendering here, but I wonder whether this is efficient or not? Seems that the cost of a `switch` has to be performed for each and every color.

Reply:
> I wouldn't worry about the performance until we have traces that show performance here is a bottleneck.

You are right!

> we could always up this function to operate over an entire vector at once

That is what I was thinking :)